pkg/apis/serving/k8s_validation_test.go

```diff
@@ -513,6 +507,138 @@ func TestPodSpecMultiContainerValidation(t *testing.T) {
 	}
 }
 
+func TestPodSpecFeatureValidation(t *testing.T) {
```

Comment:
This iterate each podSpec feature running four tests against each feature. It makes adding new tests for new podSpec features easy and reduces the amount of duplicate test code. If this is acceptable I can move the test for FieldRef into this too. If the preference is to duplicate the code for each of these for clarity I can break them out into a test function for each one. ty.